File: backend/app/schemas/evidence.py
# backend/app/schemas/evidence.py
from datetime import datetime
from pydantic import BaseModel


class EvidenceRead(BaseModel):
    id: int
    project_id: int
    uploaded_by: int
    uploader_name: str
    uploader_role: str
    filename: str
    mime_type: str
    extension: str
    size_bytes: int
    sha256: str
    description: str | None
    created_at: datetime
    download_url: str | None = None

    model_config = {"from_attributes": True}


class EvidenceUpdateInput(BaseModel):
    description: str | None = None

File: frontend/src/types/evidence.ts
// frontend/src/types/evidence.ts
export interface Evidence {
  id: number
  project_id: number
  uploaded_by: number
  uploader_name: string
  uploader_role: string
  filename: string
  mime_type: string
  extension: string
  size_bytes: number
  sha256: string
  description: string | null
  created_at: string
  download_url: string | null
}

export interface EvidenceUploadInput {
  file: File
  description?: string | null
}
